Dockerfile

```diff
@@ -1,5 +1,24 @@
 # This is a multi-stage Docker build
 
+# Create the downloads collection
+FROM python:3.6-alpine as collection
+
+WORKDIR /source/
+
+COPY . /source/
+RUN pip install -r requirements.txt
+# Validate that what was installed was what was expected
+RUN pip freeze 2>/dev/null | grep -v "deployer" > requirements.installed \
```

Comment:
it only checks that what we wrote in requirements.txt is also really what got installed .Nothing extra. This happens for example if you bump something, and a dependency loads a new dependency. You want to notice this. So yes, pip installs the correct versions. This checks if not more got installed because of auto-dependency shit :D

We do this for all our Dockers btw; its a common pitfall :)